libraries/standard/utilities/include/iot_logging_setup.h

```diff
@@ -0,0 +1,289 @@
+ *                     4 );
+ * @endcode
+ * The code above prints something like the following:
+ * @code{c}
+ * [DEBUG][LIB_NAME][2018-01-01 12:00:00] This buffer contains:
+ * 00 01 02 03
+ * @endcode
+ */
+
+/**
+ * @def IotLogError( message  )
+ * @brief Abbreviated logging macro for stand-alone message strings at level #IOT_LOG_ERROR.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLogWithArgs( IOT_LOG_ERROR, message )
+ * @endcode
+ */
+
+/**
+ * @def IotLogErrorWithArgs( ...  )
+ * @brief Abbreviated logging macro for messages with arguments at level #IOT_LOG_ERROR.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_ERROR, ... )
+ * @endcode
+ */
+
+/**
+ * @def IotLogWarn( message  )
+ * @brief Abbreviated logging macro for stand-alone message strings at level #IOT_LOG_WARN.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_WARN, message )
+ * @endcode
+ */
+
+/**
+ * @def IotLogWarnWithArgs( ...  )
+ * @brief Abbreviated logging macro for messages with arguments at level #IOT_LOG_WARN.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_WARN, ... )
+ * @endcode
+ */
+
+/**
+ * @def IotLogInfo( message  )
+ * @brief Abbreviated logging macro for stand-alone message strings at level #IOT_LOG_INFO.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_INFO, message )
+ * @endcode
+ */
+
+/**
+ * @def IotLogInfoWithArgs( ...  )
+ * @brief Abbreviated logging macro for messages with arguments at level #IOT_LOG_INFO.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_INFO, ... )
+ * @endcode
+ */
+
+/**
+ * @def IotLogDebug( message  )
+ * @brief Abbreviated logging macro for stand-alone message strings at level #IOT_LOG_DEBUG.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_DEBUG, message )
+ * @endcode
+ */
+
+/**
+ * @def IotLogDebugWithArgs( ...  )
+ * @brief Abbreviated logging macro for messages with arguments at level #IOT_LOG_DEBUG.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_DEBUG, ... )
+ * @endcode
+ */
+
+#if !defined( IotLog )
+    #error "Please define IotLog logging interface."
+#endif
+
+/* Check that LIBRARY_LOG_LEVEL is defined and has a valid value. */
+#if !defined( LIBRARY_LOG_LEVEL ) ||           \
+    ( ( LIBRARY_LOG_LEVEL != IOT_LOG_NONE ) && \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_ERROR ) &&  \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_WARN ) &&   \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_INFO ) &&   \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_DEBUG ) )
+    #error "Please define LIBRARY_LOG_LEVEL as either IOT_LOG_NONE, IOT_LOG_ERROR, IOT_LOG_WARN, IOT_LOG_INFO, or IOT_LOG_DEBUG."
+/* Check that LIBRARY_LOG_NAME is defined and has a valid value. */
+#elif !defined( LIBRARY_LOG_NAME )
+    #error "Please define LIBRARY_LOG_NAME."
+#else /* if !defined( LIBRARY_LOG_LEVEL ) || ( ( LIBRARY_LOG_LEVEL != IOT_LOG_NONE ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_ERROR ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_WARN ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_INFO ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_DEBUG ) ) */
```

Comment:
Nit: This very long comment, IMO, is making code unreadable.

Reply:
Removed. I think it was auto-populated by uncrustify when I ran it earlier, However, removing the comment explicitly didn't re-populate it.